src/systems/ec/implementations/react/packages/list/src/OrderedListWithData.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import OrderedList from './OrderedList';
+import OrderedListItem from './OrderedListItem';
+
+const OrderedListWithData = ({ items, className, ...props }) => {
+  return (
+    <OrderedList {...props} className={className}>
+      {items.map(item => {
+        if (item.nested) {
+          return (
+            <Fragment key={item.label}>
```

Comment:
I think it could be compacted like:
```
<OrderedListItem key={item.label}>
  {item.label}
  {item.nested && <OrderedListWithData items={item.nested} />}
</OrderedListItem>
```

end result:
```
const OrderedListWithData = ({ items, className, ...props }) => {
  return (
    <OrderedList {...props} className={className}>
      {items.map(item => (
        <OrderedListItem key={item.label}>
          {item.label}
          {item.nested && <OrderedListWithData items={item.nested} />}
        </OrderedListItem>
      ))}
    </OrderedList>
  );
};
```
would it work too?